### PR TITLE
Added new cart functionality, fixed bugs

### PR DESCRIPTION
**Pull request by Alegzandru**

### CODE REVIEW

File: src/components/Menu/CategoryProducts/CategoryProducts.tsx [hunk 1]
```diff
@@ -25,6 +26,29 @@ type Props = {
...
+    return 0
+  } else{
+    if (first.nameru < second.nameru) {
+      return -1
+    }
+    if (first.nameru > second.nameru) {
+      return 1
+    }
+    return 0
+  }
```
Comment (dodonAurel) [inline src/components/Menu/CategoryProducts/CategoryProducts.tsx:49]: I guess it looks nicer this:
```
    const getName = (product: Product) => product[ro ? 'name' : 'nameru']

    if (getName(first)< getName(second)) {
      return -1
    }

    if (getName(first) > getName(second)) {
      return 1
    }

    return 0
```

File: src/components/Modal/CartModal/CartModal.tsx [hunk 2]
```diff
@@ -1,39 +1,97 @@
...
+          return 'manager.mister.'
+        case 'Malina Mică' :
+          return 'manager.mister.'
+        case 'Râșcani' :
+          return 'rascanovca.mr.'
+        case 'Botanica' :
+          return 'botanica.mr.'
+        default :
+          return 'manager.mister.'
+      }
```
Comment (dodonAurel) [inline src/components/Modal/CartModal/CartModal.tsx:80]: I think `lookup object` > `switch statement` for simple cases.
I would add in const:
```
const TAKEAWAY_LOCATIONS = {
    'Telecentru': 'manager.mister.',
    'Malina Mică': 'manager.mister.',
    ...
}
```
and use it in code like this: `TAKEAWAY_LOCATIONS[location] `

File: src/components/Modal/CartModal/CartModal.tsx [hunk 3]
```diff
@@ -1,39 +1,97 @@
...
 
   const isThroughDelivery = orderType === ORDER_TYPE.delivery
 
-  const CurrentShowedBlock = orderType ? CART_FORM_COMPONENTS[orderType] : null
-
-  const orderTypeRadio = ({ text, type }: typeof ORDER_TYPE_BUTTONS[0], index: number) =>
-    <RadioButton key={index} text={text} selected={orderType === type} onChange={() => setOrderType(type)} />
+  const CurrentShowedBlock = isValidLocal ? null : orderType ? CART_FORM_COMPONENTS[orderType] : null
+  const CurrentShowedInput = isValidLocal ? ORDER_FROM_INPUTS[ORDER_FROM.local] : ORDER_FROM_INPUTS[ORDER_FROM.in_afara_localului]
+  const CurrentShowedForm = isValidLocal ? ORDER_FROM_FORMS[ORDER_FROM.local] : ORDER_FROM_FORMS[ORDER_FROM.in_afara_localului]
```
Comment (dodonAurel) [inline src/components/Modal/CartModal/CartModal.tsx:94]: add ternary in barces, like this:
```
const { local, in_afara_localului } = ORDER_FROM
...
const CurrentShowedForm = ORDER_FROM_FORMS[isValidLocal ? local : in_afara_localului]
```
and same for `CurrentShowedInput `

File: src/constants/forms.ts [hunk 4]
```diff
@@ -34,52 +36,134 @@ export const VALIDATIONS = {
...
+    text: 'Card',
+    type: ORDER_PAYMENT_METHODS.card,
+  },
+]
+
+export const ORDER_TYPE_BUTTONS = [
+  // {
+  //   text: 'Livrare prin curier',
+  //   type: ORDER_TYPE.delivery,
+  // },
```
Comment (dodonAurel) [inline src/constants/forms.ts:79]: remove comments

File: src/types/index.ts [hunk 5]
```diff
@@ -78,12 +79,21 @@ export type Drinks = WithText
 
 export type Additive = Topping | Without | Drinks
 
+// export type Without = {
+//   text: string
+//   textru: string
+// }
```
Comment (dodonAurel) [inline src/types/index.ts:85]: remove comments

File: src/types/index.ts [hunk 6]
```diff
@@ -78,12 +79,21 @@ export type Drinks = WithText
...
   topping: Map<string, {
     text: string
+    textru: string
     price: number
   }>
-  without: Set<string>
+  without: Map<string, {
+    text: string
+    textru: string
+  }>
```
Comment (dodonAurel) [inline src/types/index.ts:96]: are you sure that those are needed, I mean about Map and Set, cause initially the logics at toppingManager was with Set/Map which was removed when I added context?

File: src/utils/products.ts [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+      if (first.textru < second.textru) {
+        return -1
+      }
+
+      if (first.textru > second.textru) {
+        return 1
+      }
+
+      return 0
+  }
```
Comment (dodonAurel) [inline src/utils/products.ts:99]: there could be added the logics from the above situation

File: src/utils/products.ts [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+
+export const getTipulComenziiMail = (payment: string) => {
+  switch(payment){
+    case 'takeaway' :
+      return 'Preluare din local'
+    case 'delivery' :
+      return 'Livrare'
+    default:
+      return 'În local'
+  }
```
Comment (dodonAurel) [inline src/utils/products.ts:198]: also there I would prefer object lookup

**Review by dodonAurel — CHANGES_REQUESTED** [review]

File: src/components/Modal/CartModal/CartModal.tsx (hunk 3 above)
Comment (Alegzandru) [reply src/components/Modal/CartModal/CartModal.tsx:94]: Da pentru ce sa fac const { local, in_afara_localului } = ORDER_FROM?
Nu pot prosta sa adaug ca string in 
ORDER_FROM_FORMS[isValidLocal ? 'local' : 'in_afara_localului']

File: src/types/index.ts (hunk 6 above)
Comment (Alegzandru) [reply src/types/index.ts:96]: So i can delete this?

**Pull request merged**